File: mysql_examples/recipes/api/01_connect/connect2.pl
#!/usr/bin/perl
# connect2.pl - connect to the MySQL server, showing how to specify
# a port number or Unix domain socket path explicitly.

use strict;
use warnings;
use DBI;

{ # begin scope

#@ _FRAG1_
my $dsn = "DBI:mysql:host=mysql.example.com;database=cookbook"
            . ";port=3307";
#@ _FRAG1_
my $dbh = DBI->connect ($dsn, "cbuser", "cbpass");
if (!$dbh)
{
  warn "Cannot connect to server\n";
}
else
{
  print "Connected\n";
  $dbh->disconnect ();
  print "Disconnected\n";
}

} # end scope

{ # begin scope

#@ _FRAG2_
my $dsn = "DBI:mysql:host=localhost;database=cookbook"
            . ";mysql_socket=/var/tmp/mysql.sock";
#@ _FRAG2_
my $dbh = DBI->connect ($dsn, "cbuser", "cbpass");
if (!$dbh)
{
  warn "Cannot connect to server\n";
}
else
{
  print "Connected\n";
  $dbh->disconnect ();
  print "Disconnected\n";
}

} # end scope

File: mysql_examples/recipes/api/01_connect/connect2.rb
#!/usr/bin/ruby -w
# connect2.rb - connect to the MySQL server, showing how to specify
# a port number or Unix domain socket path explicitly.

require "dbi"

begin
#@ _FRAG1_
  dsn = "DBI:Mysql:host=mysql.example.com;database=cookbook" +
          ";port=3307"
#@ _FRAG1_
  puts "Connect using DSN = #{dsn}"
  dbh = DBI.connect(dsn, "cbuser", "cbpass")
  puts "Connected"
rescue
  puts "Cannot connect to server"
else
  dbh.disconnect
  puts "Disconnected"
end

begin
#@ _FRAG2_
  dsn = "DBI:Mysql:host=localhost;database=cookbook" +
          ";socket=/var/tmp/mysql.sock"
#@ _FRAG2_
  puts "Connect using DSN = #{dsn}"
  dbh = DBI.connect(dsn, "cbuser", "cbpass")
  puts "Connected"
rescue
  puts "Cannot connect to server"
else
  dbh.disconnect
  puts "Disconnected"
end
